scalalib/src/ScalaModule.scala

```diff
@@ -283,7 +304,7 @@ trait ScalaModule extends JavaModule { outer =>
 
       packageWithZinc(
         Seq("-d", javadocDir.toNIO.toString),
-        allSourceFiles().map(_.path.toString),
+        processedDocSources().map(_.path.toString),
```

Comment:
The doesn't look right. `processedDocSources` seem to contain directories, whereas `allSourceFiles` were concrete source files.

Reply:
I also thook about docSources but this method already exists and it isn't the same feature. According to scaladocs, `docSources` is for static (not processed by scaladoc) files.

What if we have, for some obscure reason, a tasty file and we just want to have it in our doc without processing it ?

Reply:
This doesn't address my comment.

Reply:
My edit didn't pass. I'll re-edit the message

Reply:
No, `processDocSources` is meant to be overridden as configuration. We have to handle it here. Just walk all dirs returned from `processedDocSources` and use all source files.